### DISCUSSION TITLE
如何调教数据集和字典才能达到对七段式数码管的更好识别效果？

### DISCUSSION THREAD

**Discussion by rainote2020** [PaddlePaddle/PaddleOCR#14214] [General]
简单来说，我有一个3000张左右的七段式数码管的数据集。这组数据集中大部分图像都是不清晰的。并且我只需要考虑识别就够了。下图这个是比较清晰的，只不过颜色区分度不高。
![0302_002222_50 0](https://github.com/user-attachments/assets/f3d9e5ed-8816-4734-8fb3-ac864bc012ce)

我尝试在V4的预训练英文识别模型基础上训练我自己的数据集。但是过程中我遇到了如下让人费解的问题：
1. 原图中有部分数码管，它的6长得像b，而9长得像q。我尝试把他们标注为6和9，但是得到的模型测试结果惨不忍睹。而将其标注为b和q，模型的结果反而非常好。这是为什么？如果是说b和6、9和q长得不像，那不同数码管中相同数字长得也不是很像，怎么就上面说的情况影响那么大呢？
2. 原图中有字母A，但是七段式数码管的A长得有点像R。我是将其标注为R还是标注为A呢？测试下来两者的loss差不多，但是训练完成的模型出错的情况不一样(数量倒是相差不大)。这又是为什么？
3. 我的数据集基本上是不会再改变了，如何进一步提升精确度，乃至能达到100%？

求大佬帮忙解答一下疑惑。

**Comment by UserWangZz** [comment]
字典覆盖图片中所有的情况就行吧，这样子可以减少混淆字符带来的性能降低。但是需要考虑后续是否会有字符集的扩展需求

**Reply by rainote2020** [reply]
> 字典覆盖图片中所有的情况就行吧，这样子可以减少混淆字符带来的性能降低。但是需要考虑后续是否会有字符集的扩展需求

我发现有时候训练出的模型虽然出错的数量差不多，但是出错的情况不太相同。
那么我如果为了更高的准确度，能不能对同一组数据集训练多次得到多个模型，然后考虑这几个模型出错的地方可能不一样。这样让他们同时推理，输出结果最多的那个为最终结果。

我问了问ai，它表示这个方法可行。但是不清楚有没有人具体做过。

**Reply by UserWangZz** [reply]
感觉是可以的，但是考虑后处理会比较复杂，并且多个模型推理也需要考虑如何并行，是一个思路不过实现起来可能比较复杂
